commit 8798cdb6a94dcddd97af2d6267b62c80a1608240
Author: Yun Peng <pcloudy@google.com>
Date:   Wed Jan 5 17:48:13 2022 +0100

    Add MODULE.bazel for examples and clear the WORKSPACE file

diff --git a/examples/MODULE.bazel b/examples/MODULE.bazel
new file mode 100644
index 000000000..20dbae8cc
--- /dev/null
+++ b/examples/MODULE.bazel
@@ -0,0 +1,7 @@
+bazel_dep(name = "protobuf", repo_name = "com_google_protobuf", version = "")
+
+local_path_override(
+  module_name = "protobuf",
+  path = "..",
+)
+
diff --git a/examples/WORKSPACE b/examples/WORKSPACE
index fb36639ae..9fb7e4dc9 100644
--- a/examples/WORKSPACE
+++ b/examples/WORKSPACE
@@ -1,31 +1,2 @@
 workspace(name = "com_google_protobuf_examples")
 
-load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
-
-# This com_google_protobuf repository is required for proto_library rule.
-# It provides the protocol compiler binary (i.e., protoc).
-#
-# We declare it as local_repository so we can test changes
-# before they get merged. You'll want to use the following instead:
-#
-# http_archive(
-#     name = "com_google_protobuf",
-#     strip_prefix = "protobuf-master",
-#     urls = ["https://github.com/protocolbuffers/protobuf/archive/master.zip"],
-# )
-local_repository(
-    name = "com_google_protobuf",
-    path = "..",
-)
-
-# Similar to com_google_protobuf but for Java lite. If you are building
-# for Android, the lite version should be preferred because it has a much
-# smaller code size.
-local_repository(
-    name = "com_google_protobuf_javalite",
-    path = "..",
-)
-
-load("@com_google_protobuf//:protobuf_deps.bzl", "protobuf_deps")
-
-protobuf_deps()
